packages/mapper/src/mapper.ts

```diff
@@ -1861,29 +1929,41 @@ export default class Mapper {
     explainable._verbosity = verbosity;
   }
 
-  explainable_find(explainable: Explainable, query?: any, projection?: any): any {
-    this._emitExplainableApiCall(explainable, 'find', { query, projection });
-
-    // TODO: turn ExplainableCursor in a proper type.
-    const cursor = explainable._collection.find(query, projection);
-    cursor.shellApiType = (): string => 'ExplainableCursor';
+  // TODO: turn this into proper types and constructors in shell api.
```

Comment:
This is done right? Or are there other types

Reply:
Want to do that in a separate PR, i have to fix the output anyway. We probably need another proxy `new ExplainableCursor(cursor, verbosity)`, for which  we keep methods in sync with the original cursor, was thinking to use anchors in YAML for that.